MassMessage/index.ts

```diff
@@ -16,47 +17,55 @@ const timerTrigger: AzureFunction = async function (context: Context, myTimer: a
 
     await MongoConnect()
 
-    const date = new Date()
+    const curDate = new Date()
     //avoiding the use of times
-    date.setHours(0)
-    date.setMinutes(0)
-    date.setSeconds(0)
-    date.setMilliseconds(0)
+    curDate.setHours(0)
+    curDate.setMinutes(0)
+    curDate.setSeconds(0)
+    curDate.setMilliseconds(0)
+    // passing this date around so we don't need to repeatedly set times to 0
 
-    //set to two months ago
-    date.setMonth(date.getMonth() - 2)
-
-    const inactive = await messageInactiveUsers(date)
+    const inactive = await messageInactiveUsers(curDate)
     context.log(inactive)
-    const catchup = await messageCompletedUsers(date, context)
+    const catchup = await messageCompletedUsers(curDate, context)
     context.log(catchup)
 }
 
-//gets users where last activity was exactly 2 months ago
+//gets users where last activity was exactly 2 weeks ago
 const messageInactiveUsers = async function (date: Date) {
+    //set to two weeks ago
+    const twoWeeks = new Date(date);
+    twoWeeks.setDate(twoWeeks.getDate() - 14);
+    
     const allUsers = await UserState.find({
-        lastActivity: date,
+        lastActivity: twoWeeks,
     })
     allUsers.forEach((user) => {
-        sendInactiveMessage(user.userId, date.toDateString())
+        sendInactiveMessage(user.userId, twoWeeks.toDateString(), user.lowData)
+        //hardcoded to welcome message
+        user.currMessage = Mongoose.Types.ObjectId("6022178429efc055c8e74e50");
     })
 }
 
 //gets users where they completed a module 2 months ago
 const messageCompletedUsers = async function (date: Date, context: Context) {
-    const prevDay = new Date(date.toDateString())
+    //set to two months ago
+    const twoMonths = new Date(date);
+    twoMonths.setMonth(twoMonths.getMonth() - 2);
+
+    const prevDay = new Date(twoMonths)
     prevDay.setDate(prevDay.getDate() - 1)
 
     const allUsers = await UserState.find({
         moduleCompletionTime: {
-            $gte: prevDay,
-            $lt: date,
+            $gt: prevDay,
+            $lt: twoMonths,
         },
     })
     allUsers.forEach((user) => {
```

Comment:
Based on all of the above comments, we may need to rethink how we do this array logic as well. The outer forEach is probably fine, but the easiest way to do it to make one pass through all of the users and sort them into buckets of module 0 complete, module 1 complete, module 2 complete, etc. and then send that specific associated message from the fixed messages based on which one is complete.

The one edge case that this wouldn't work for is if a user completes 2 modules at the same time, but we'll probably have to write something to bump up the time of completion for the later modules by 24 hours and resend a diagnostic quiz 24 hours later, but I need to ask the nonprofit about this.